src/cpprealm/experimental/macros.hpp

```diff
@@ -307,41 +318,103 @@ rbool managed<std::optional<type>>::operator op(const std::optional<type>& rhs)
+                 std::apply([&](auto &&...ptr) { \
+                     std::apply([&](auto &&..._name) { \
+                     ((*this.*ptr).assign(&m_obj, &m_realm, m_obj.get_table().get_column_key(_name)), ...); \
+                     }, managed_pointers_names); \
+                 }, managed_pointers());                                                                \
+             }                                                                                       \
         } \
         managed& operator=(managed&& other) { \
-             m_obj = std::move(other.m_obj); \
-             m_realm = std::move(other.m_realm);                                                   \
-            std::apply([&](auto && ...ptr) { \
-                std::apply([&](auto&& ..._name) { \
-                ((*this.*ptr).assign(&m_obj, &m_realm, m_obj.get_table().get_column_key(_name)), ...); \
-                }, managed_pointers_names); \
-            }, managed_pointers()); \
+            m_obj = std::move(other.m_obj); \
+            m_realm = std::move(other.m_realm);                                                   \
+            m_prepare_for_query = std::move(other.m_prepare_for_query);                                     \
+            if (m_prepare_for_query) {                                                                                       \
+                auto schema = m_realm.schema().find(other.schema.name);                                  \
+                auto group = m_realm.read_group();                                                   \
+                auto table_ref = group.get_table(schema.table_key());                                  \
+                std::apply([&](auto &&...ptr) {                                                        \
+                   std::apply([&](auto &&..._name) {                                                   \
+                     ((*this.*ptr).prepare_for_query(&m_realm, table_ref, _name), ...);                \
+                   }, managed_pointers_names);                                                         \
+                }, managed_pointers());                                                                 \
+                } else {                                                                                      \
+                   std::apply([&](auto &&...ptr) { \
+                        std::apply([&](auto &&..._name) { \
+                        ((*this.*ptr).assign(&m_obj, &m_realm, m_obj.get_table().get_column_key(_name)), ...); \
+                    }, managed_pointers_names); \
+                }, managed_pointers());                                                                \
+            }  \
              return *this;\
-        } \
+        }                                                                                          \
+        static managed prepare_for_query(const internal::bridge::realm& r) {                       \
```

Comment:
Can you give more detail for what these macro changes are?

Reply:
If you have a link property you need to propagate the `prepare_for_query` logic down the chain.